### PR TITLE
Minimal imports

### LINKED ISSUE #2013: cloudpickle requirement
``multiprocessing`` imports  ``cloudpickle`` since deb74696e859c. Shouldn't  ``cloudpickle`` be in ``setup.py``'s ``install_req``?

### PR DESCRIPTION
**Pull request by jcrist**
- Fix to make cloudpickle not a required import
- Move appveyor scripts to `continuous_integration/appveyor` folder
- Split `.travis.yml` into subscripts, move to `continuous_integration/travis` folder
- Add tests that imports with minimal dependencies work

Fixes #2013

### CODE REVIEW

**Comment by jcrist** [comment]
This *could* also be done using pip, but in my tests it's significantly slower. Using pip would be more robust though, as we wouldn't have two dependency lists to keep in sync. Hoping that the added comment to `setup.py` is sufficient.

File: continuous_integration/travis/install.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    pip install backports.lzma mock
+    pip install git+https://github.com/Blosc/castra
+fi
+
+if [[ $PYTHON == '3.5' ]]; then
+    pip install git+https://github.com/dask/fastparquet
+fi
+
+# Install dask
+pip install --no-deps -e .[complete]
```
Comment (mrocklin) [inline continuous_integration/travis/install.sh]: I like that this makes it easy to create a very similar environment on local hardware, which is sometimes nice if you want to recreate an issue that is happening on travis-ci

**Comment by jcrist** [comment]
Not sure why I'm seeing consistent timeouts on matrix builds 2 and 4. Hmmm.

**Comment by jcrist** [comment]
Due to some hard to debug incompatibilities between conda-forge and the default channel, this took a bit, but it finally works. Also simplified our dependencies a bit, we had some old cruft that had accumulated. Merging.

**Pull request merged**